doc/headset.md

```diff
@@ -0,0 +1,430 @@
+```
+
+Params:
+* `isMuted: boolean` - value to determine if the device is now muted or unmuted
+
+Returns: a Promise containing `void`
+
+#### `setHold()`
+Function that calls the headset library's `setHold(conversationId, isHeld)` function. This signals the headset device to switch on or off (depending the value of `isHeld`) the hold call button's light to show the call has been held or resumed respectively
+
+Declaration:
+``` ts
+    setHold(conversationId: string, isHeld: boolean): Promise<void>;
+```
+
+Params:
+* `conversationId: string` - The conversationId of the call that needs to be held or resumed
+* `isHeld: boolean` - value to determine if the device is now on hold or resumed
+
+Returns: a Promise containing `void`
+
+--------
+
+### Events
+The SDK Headset Utility does not explicitly emit events itself. It uses RxJS observables (https://rxjs.dev/guide/observable) to emit the events which are then subscribed to within the consuming app. It listens for changes and fires functions that correspond to the events
+
+### `deviceAnsweredCall`
+Event emitted when a user presses the answer call button during an incoming call on their selected device. The event includes the event `name` as it is interpretted by the headset and a collection of items that may help with logging (`event`). It can also potentially have a `code` that corresponds to the event.
+
+Declaration:
+``` ts
+    sdk.headset.headsetEvents.subscribe(event: {
+        event: 'deviceAnsweredCall',
+        payload: {
+            name: string,
+            code?: string,
+            event: { `containing various items mostly for logging purposes` }
+        }
+    } => {
+        if (event.event === 'deviceAnsweredCall') {
+            sdk.acceptPendingSession();
+        }
+    })
+```
+
+Value of event:
+* `event: EmittedHeadsetEvents` - string value emitted by the headset library to determine what event had just occurred
+* `payload:` {
+    `name: string` - Name of the recent event as interpretted by the headset device
+    `event`: { containing various items mostly for logging purposes}
+    `code?: string` - Optional: A string value of a number that represents the action that was just taken. Not all vendors supply a code which is why it is only optional
+}
+
+
+#### `deviceEndedCall`
+Event emitted when a user presses the answer call button while in an active call on their selected device. The event includes the event `name` as it is interpretted by the headset and a collection of items that may help with logging (`event`). It can also potentially have a `code` that corresponds to the event.
+
+Declaration:
+``` ts
+    sdk.headset.headsetEvents.subscribe(event: {
+        event: 'deviceEndedCall',
+        payload: {
+            name: string,
+            event: { `containing various items mostly for logging purposes` },
+            code?: string
+        } => {
+            if (event.event === 'deviceEndedCall) {
+                sdk.endSession();
```

Comment:
Also, just for documentation sake, you may want to call this with conversationId. They _should_ to use conversationId and typescript will error on them if they don't, but not all our consumers use typescript. 

``` ts
sdk.endSession({ conversationId });
```